components/rollup_node/rollup_driver.md

```diff
@@ -0,0 +1,27 @@
+# Rollup Driver
+
+The Rollup Driver is the core code which guarantees that the canonical L1 state is accurately reflected in the EE. It does this by connecting to an L1 node, and using the pure function which maps L1 blocks to L2 block inputs as defined by [block generation][block-gen]. Once it knows the block inputs, it checks the EE to make sure that the blocks there all have the correct inputs.
+
+This is a functionality that is shared between all configurations of the rollup node, including the sequencer. Because block generation defines the L1 block -> L2 block input transformation as a pure function, it can be completely stateless and has no DB.
+
+## Basic Functionality
+
+We define the `latestL1ConfirmedL2Block` as the latest L2 block in the EE which is marked as L1-confirmed. This can be found by starting at the head of the L2 EE, and iterating backwards until we find the first L2 block which is marked as L1-confirmed.
+
+The core logic which the CL executes is as follows.
```

Comment:
I think this describes it well, but generally we can make it easier and more robust by doing the forkchoice update first, and then the execution-engine executes the reorg logic (it already implements sync and reorgs anyway). If it can't find the blocks/state from peers, the rollup-node can then provide them. This is what Peter from Geth argues for at least.